server/server_test.go

```diff
@@ -75,7 +75,7 @@ func TestNewSocketServer(t *testing.T) {
 	ret := newSocketServer(cfg, nil)
 	assert.NotEqual(t, nil, ret, "ret : isNil()")
 	assert.Equal(t, mockServer.Addr, ret.Addr, fmt.Sprintf("Addr invalide: %v != %v",
-		ret.Addr, mockServer.Addr))
+		ret.Addr, mockServer.Addr)) //nolint:staticcheck // non-nil ret is ensures by the assert above
```

Comment:
Why is this necessary for an assert statement?

Reply:
here's the linter's output:
```bash
$ golangci-lint run ./... | grep server_test.go
server/server_test.go:77:39: SA5011: possible nil pointer dereference (staticcheck)
server/server_test.go:84:5: SA5011(related information): this check suggests that the pointer can be nil (staticcheck)
server/server_test.go:79:46: SA5011: possible nil pointer dereference (staticcheck)
server/server_test.go:114:5: SA5011(related information): this check suggests that the pointer can be nil (staticcheck)
```
There is a check `if ret != nil {` after those asserts which makes the linter suspicious.  I suggest we use `require.NotNil(t, ret, "ret : isNil()")` instead of `assert.NotEqual(t, nil, ret, "ret : isNil()")` and then we can delete the nil check around `ret.Close()`:
```go
       if ret != nil {
               ret.Close()
       }
```

Reply:
removed these `//nolint` comments